playground/src/components/echo-server/echo-server.js

```diff
@@ -8,8 +8,23 @@ import Text from '@commercetools-uikit/text';
 import PrimaryButton from '@commercetools-uikit/primary-button';
 import { CodeBlock } from '@commercetools-docs/ui-kit';
 
+const initialState = {
+  isLoading: false,
+  result: null,
+};
+const reducer = (state, action) => {
+  switch (action.type) {
+    case 'loading':
+      return { isLoading: true, result: null };
+    case 'complete':
+      return { isLoading: false, result: action.payload };
+    default:
+      return state;
+  }
+};
+
 const EchoServer = () => {
-  const [result, setResult] = React.useState();
+  const [state, dispatchState] = React.useReducer(reducer, initialState);
```

Comment:
It's a bit too much here maybe but usually encapsulating helps so:

```js
const { dispatchComplete, dispatchLoading } = useLoadingStateReducer()
```

Reply:
Yeah I just wanted to keep it simple. I can change that though, if you prefer so.

Reply:
All good. Thanks for considering.